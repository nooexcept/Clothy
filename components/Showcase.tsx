import * as React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardActionArea from '@material-ui/core/CardActionArea'
import CardActions from '@material-ui/core/CardActions'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'

const useStyles = makeStyles({
    root: {
        width: '30vw',
        minWidth: 200,
        background: '#000',
    },
    media: {
        height: '55vh',
        minHeight: 150,
    },
    content: {
        backgroundColor: '#fbd46d',
    },
    actions: {
        backgroundColor: '#fbd46d',
    },
})

interface ShowcaseProps {
    img: string
    title: string
    description: string
}

export default function Showcase(props: ShowcaseProps): React.ReactElement {
    const classes = useStyles()

    return (
        <Card className={classes.root}>
            <CardActionArea>
                <CardMedia
                    className={classes.media}
                    image={props.img}
                    title={props.title}
                />
                <CardContent className={classes.content}>
                    <Typography
                        color="textPrimary"
                        gutterBottom
                        variant="h5"
                        component="h2"
                    >
                        {props.title}
                    </Typography>
                    <Typography
                        variant="body2"
                        color="textPrimary"
                        component="p"
                        noWrap
                    >
                        {props.description}
                    </Typography>
                </CardContent>
            </CardActionArea>
            <CardActions className={classes.actions} disableSpacing>
                <Button size="small">
                    <Typography color="textPrimary" variant="body1">
                        Share
                    </Typography>
                </Button>
                <Button size="small">
                    <Typography color="textPrimary" variant="body1">
                        Browse
                    </Typography>
                </Button>
            </CardActions>
        </Card>
    )
}
