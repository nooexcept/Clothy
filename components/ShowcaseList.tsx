import * as React from 'react'
import { makeStyles, createStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Zoom from '@material-ui/core/Zoom'

const useStyles = makeStyles(() =>
    createStyles({
        root: {
            flexGrow: 1,
            maxWidth: '100%',
        },
    })
)

interface ShowcaseListProps {
    showcases: Array<React.ReactElement>
}

export default function ShowcaseList(
    props: ShowcaseListProps
): React.ReactElement {
    const classes = useStyles()

    return (
        <Grid container className={classes.root} spacing={2}>
            <Grid item xs={12}>
                <Grid container justify="center" spacing={2}>
                    {props.showcases.map((showcase, i) => (
                        <Zoom
                            in
                            key={i}
                            style={{
                                transitionDelay: `${100 + 100 * i}ms`,
                            }}
                        >
                            <Grid item>{showcase}</Grid>
                        </Zoom>
                    ))}
                </Grid>
            </Grid>
        </Grid>
    )
}
