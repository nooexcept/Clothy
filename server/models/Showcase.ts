import mongoose from 'mongoose'

export interface Showcase extends mongoose.Document {
    collId: string
}

const showcaseSchema = new mongoose.Schema({
    collId: {
        type: String,
        required: true,
    },
})

export default mongoose.model<Showcase>('Showcase', showcaseSchema)
