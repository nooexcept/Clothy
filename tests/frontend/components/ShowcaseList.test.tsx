import React from 'react'
import { ShallowWrapper } from 'enzyme'
import { createShallow } from '@material-ui/core/test-utils'
import ShowcaseList from '../../../components/ShowcaseList'
import Showcase from '../../../components/Showcase'

let shallow: ReturnType<typeof createShallow>
let wrapper: ShallowWrapper

const mockedProd = {
    title: 'Ultra chill jumpers',
    description: 'If you wanna look chill, you need ultra chill jumpers',
    img: '/photo-of-woman-standing-on-staircase-3120339.jpg',
}

beforeEach(() => {
    shallow = createShallow()
    wrapper = shallow(
        <ShowcaseList
            showcases={[
                <Showcase key={0} {...mockedProd} />,
                <Showcase key={1} {...mockedProd} />,
                <Showcase key={2} {...mockedProd} />,
            ]}
        />
    )
})

describe('<ShowcaseList /> rendering', () => {
    it('should render without problems', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('should render all <Showcase /> elements specified', () => {
        expect(wrapper.find(Showcase).length).toEqual(3)
    })
})
