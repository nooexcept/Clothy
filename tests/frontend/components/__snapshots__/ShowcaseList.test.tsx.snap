// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowcaseList /> rendering should render without problems 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  justify="center"
  spacing={2}
>
  <ForwardRef(Zoom)
    in={true}
    key="0"
    style={
      Object {
        "transitionDelay": "100ms",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Showcase
        description="If you wanna look chill, you need ultra chill jumpers"
        img="/photo-of-woman-standing-on-staircase-3120339.jpg"
        key="0"
        title="Ultra chill jumpers"
      />
    </WithStyles(ForwardRef(Grid))>
  </ForwardRef(Zoom)>
  <ForwardRef(Zoom)
    in={true}
    key="1"
    style={
      Object {
        "transitionDelay": "200ms",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Showcase
        description="If you wanna look chill, you need ultra chill jumpers"
        img="/photo-of-woman-standing-on-staircase-3120339.jpg"
        key="1"
        title="Ultra chill jumpers"
      />
    </WithStyles(ForwardRef(Grid))>
  </ForwardRef(Zoom)>
  <ForwardRef(Zoom)
    in={true}
    key="2"
    style={
      Object {
        "transitionDelay": "300ms",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <Showcase
        description="If you wanna look chill, you need ultra chill jumpers"
        img="/photo-of-woman-standing-on-staircase-3120339.jpg"
        key="2"
        title="Ultra chill jumpers"
      />
    </WithStyles(ForwardRef(Grid))>
  </ForwardRef(Zoom)>
</WithStyles(ForwardRef(Grid))>
`;
